### PR TITLE
Add support for external rules

### PR DESCRIPTION
**Pull request by wooorm**
*   Add example rule, and rule object file, to
    `test/external`;
*   Add fixtures for external tests;
*   Move `mdast.js`, `mdast.min.js` to `mdast-lint.js`,
    `mdast-lint.min.js`;
*   Add docs for external rules.

Closes GH-14.

### CODE REVIEW

**Comment by wooorm** [comment]
@zcei What do you think?

File: lib/index.js [hunk 1]
```diff
@@ -86,6 +96,79 @@ function attachFactory(id, rule, options) {
...
 
 /**
+ * Require an external.  Checks, in this order:
+ *
+ * -  `$cwd/$pathlike`;
+ * -  `$cwd/$pathlike.js`;
+ * -  `$cwd/node_modules/$pathlike`;
+ * -  `$pathlike`.
+ *
+ * Where `$package` is an ancestral package directory and
```
Comment (zcei) [inline lib/index.js]: You mean `$pathlike` here, or more likely $package the lines above, right?

File: lib/index.js [hunk 2]
```diff
@@ -14,10 +14,20 @@
 
 var range = require('mdast-range');
 var zone = require('mdast-zone');
-var rules = require('./rules');
+var defaultRules = require('./rules');
```
Comment (zcei) [inline lib/index.js]: What do you think about `nativeRules` here, as they define no standards, but are the native rules of the package?

**Comment by zcei** [comment]
That's a pretty nice PR! :+1: 
Awesome, that you've tackled it so fast :rocket:

**Pull request merged**